packages/datadog-instrumentations/src/child_process.js

```diff
@@ -19,32 +19,14 @@ const names = ['child_process', 'node:child_process']
 // child_process and node:child_process returns the same object instance, we only want to add hooks once
 let patched = false
 
-function throwSyncError (error) {
-  throw error
-}
-
-function returnSpawnSyncError (error, context) {
-  context.result = {
-    error,
-    status: null,
-    signal: null,
-    output: null,
-    stdout: null,
-    stderr: null,
-    pid: 0
-  }
-
-  return context.result
-}
-
 names.forEach(name => {
   addHook({ name }, childProcess => {
     if (!patched) {
       patched = true
       shimmer.massWrap(childProcess, execAsyncMethods, wrapChildProcessAsyncMethod(childProcess.ChildProcess))
-      shimmer.wrap(childProcess, 'execSync', wrapChildProcessSyncMethod(throwSyncError, true))
-      shimmer.wrap(childProcess, 'execFileSync', wrapChildProcessSyncMethod(throwSyncError))
-      shimmer.wrap(childProcess, 'spawnSync', wrapChildProcessSyncMethod(returnSpawnSyncError))
+      shimmer.wrap(childProcess, 'execSync', wrapChildProcessSyncMethod(true))
+      shimmer.wrap(childProcess, 'execFileSync', wrapChildProcessSyncMethod())
+      shimmer.wrap(childProcess, 'spawnSync', wrapChildProcessSyncMethod())
```

Comment:
spawnSync does not throw an error when a command is wrong, it returns a value with the result, for example, if I execute this method:
```javascript
function test (cmd) {
  try {
    const result = childProcess.spawnSync('noop'
    return result
  } catch (e) {
    console.error('ERROR!', e)
    return null
  }
}
```
catch code is not executed, this command returns an object and `console.error('ERROR!', e)` is not executed.

```
> test('noop')
{
  error: Error: spawnSync noop ENOENT
      at Object.spawnSync (node:internal/child_process:1124:20)
      at Object.spawnSync (node:child_process:876:24)
      at teste (REPL83:3:31)
      at REPL96:1:1
      at Script.runInThisContext (node:vm:122:12)
      at REPLServer.defaultEval (node:repl:594:29)
      at bound (node:domain:432:15)
      at REPLServer.runBound [as eval] (node:domain:443:12)
      at REPLServer.onLine (node:repl:924:10)
      at REPLServer.emit (node:events:526:35) {
    errno: -2,
    code: 'ENOENT',
    syscall: 'spawnSync noop',
    path: 'noop',
    spawnargs: []
  },
  status: null,
  signal: null,
  output: null,
  pid: 0,
  stdout: null,
  stderr: null
}
```

Reply:
I reverted my changes, will add some tests for spawnSync